agent/csi/plugin_test.go

```diff
@@ -7,51 +7,34 @@ import (
 	"context"
 
 	"github.com/container-storage-interface/spec/lib/go/csi"
-	"github.com/docker/swarmkit/api"
 )
 
 const driverName = "testdriver"
 
-// newPluginFromClients creates a new node plugin using the provided CSI RPC
-// clients.
-func newPluginFromClients(name string, nodeClient csi.NodeClient, nodeId string) *NodePlugin {
-	return &NodePlugin{
-		name:       name,
-		nodeClient: nodeClient,
-		nodeID:     nodeId,
-	}
-}
-
 var _ = Describe("Plugin Node", func() {
 	var (
-		nodePlugin *NodePlugin
-
 		nodeClient *fakeNodeClient
 
-		nodeInfo *api.NodeCSIInfo
+		nodeInfo *csi.NodeGetInfoResponse
 
 		err error
 	)
 
 	BeforeEach(func() {
 
-		nodeClient = newFakeNodeClient()
+		nodeClient = newFakeNodeClient(true, "nodeid1")
 
-		nodePlugin = newPluginFromClients(driverName, nodeClient, "1")
 	})
 
 	JustBeforeEach(func() {
-		nodeInfo, err = nodePlugin.NodeGetInfo(context.Background())
+		nodeInfo, err = nodeClient.NodeGetInfo(context.Background(), &csi.NodeGetInfoRequest{})
```

Comment:
This doesn't really test the plugin, just the fake. Instead, you need create a `NodePlugin` object, and then set its node client to be this fake.